### PR TITLE
Use cached namespaces during validation

### LINKED ISSUE #528: [Bug]: PyNWB validator not using cached schema
### What happened?

Using the dev branch, the PyNWB validator is not using cached extensions during validation. I think that is because we are using the `validate(io=io)` method instead of the `validate(paths=paths)` method. 

https://github.com/NeurodataWithoutBorders/nwbinspector/blame/ba7d845b255e8095c72978e91b2c014675ef55d6/src/nwbinspector/_nwb_inspection.py#L277

This appears to have been changed recently to support validation with Zarr.

See also https://github.com/NeurodataWithoutBorders/pynwb/issues/1777

I will look into solutions. We should add a test for this.

### Operating System

macOS

### Python Version

3.12

### Were you streaming with ROS3?

No

### Package Versions

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/NeurodataWithoutBorders/nwbinspector/blob/dev/.github/CODE_OF_CONDUCT.rst)
- [X] Have you ensured this bug was not already [reported](https://github.com/neurodatawithoutborders/nwbinspector/issues)?
- [X] To the best of your ability, have you ensured this is a bug within the code that checks the NWBFile, rather than a bug in the NWBFile reader (e.g., [PyNWB](https://github.com/NeurodataWithoutBorders/pynwb) or [MatNWB](https://github.com/NeurodataWithoutBorders/matnwb))?

### PR DESCRIPTION
**Pull request by stephprince**
## Motivation

Fix #528. 

Recent Zarr-related updates switched the inspector from using `pynwb.validate(paths=paths)` to `pynwb.validate(io=io)`,  resulting in the inspector not using cached namespaces during validation. We will temporarily revert this change, since pynwb validation of Zarr NWB files is not yet fully supported anyway due to pending issues that need to be addressed in hdmf/pynwb.

Since this is more of a bug on the pynwb side that using the `io` argument ignores the `use_cached_namespaces=True` setting in the validator (and will be updated in the next major pynwb release), I'm not sure if there are really any specific tests to add here.

While addressing this issue, I added a couple of other small updates to the Zarr support here:
1. Passing the path of an NWB file Zarr folder to the CLI (instead of the parent folder) would result in 0 files being inspected, due to some conditionals in `inspect_all` that detected whether the path provided was a directory of NWB files or an NWB file.
2. The `test_inspect_all_parallel` tests had been indented and were not running

### CODE REVIEW

File: src/nwbinspector/_nwb_inspection.py [hunk 1]
```diff
@@ -126,7 +126,9 @@ def inspect_all(
     if progress_bar_options is None:
         progress_bar_options = dict(position=0, leave=False)
 
-    if in_path.is_dir():
+    if in_path.is_dir() and (in_path.match("*.nwb*")) and (in_path / ".zgroup").exists():
```
Comment (stephprince) [inline src/nwbinspector/_nwb_inspection.py]: I'm not sure if `(in_path / ".zgroup").exists()` is the appropriate check here for a Zarr NWB file or if it is even necessary

**Comment by stephprince** [comment]
@rly you mentioned you had a related test file, can you let me know if this solves the issue or share the file?

File: src/nwbinspector/_nwb_inspection.py (hunk 1 above)
Comment (rly) [reply src/nwbinspector/_nwb_inspection.py]: Can we use https://github.com/hdmf-dev/hdmf-zarr/blob/dev/src/hdmf_zarr/backend.py#L78 ?

**Comment by stephprince** [comment]
@rly this should be ready for you to test with your example file and review

**Review by rly — APPROVED** [review]

**Pull request merged**